spec_uploader.py

```diff
@@ -120,7 +120,7 @@ def upload_specs(
             if "sandbox" in env:  # we don't want to publish stuff for sandbox
                 continue
             print(f"checking if this spec is on the portal in {env}")
-            ns_spec_name = f"{spec_name}-{env}"
+            ns_spec_name = spec_name if args["--pull-request"] else f"{spec_name}-{env}"
```

Comment:
`args` isn't in this namespace lexically -- if it is there then it's defined by coincidence of having been run as a script.

To avoid weird bugs (such as importing the function, and it failing to find `args`), you should add a new argument to the function and pass it in as an arg, such as is done from line 154 onward.